src/triage/tracking.py

```diff
@@ -0,0 +1,282 @@
+import datetime
+import platform
+import getpass
+import os
+import requests
+import subprocess
+import logging
+from functools import wraps
+from triage.util.db import scoped_session
+from contextlib import contextmanager
+from triage import __version__
+
+
+from triage.component.results_schema import ExperimentRun, ExperimentRunStatus
+
+
+def infer_git_hash():
+    """Attempt to infer the git hash of the repository in the current working directory
+
+    Returns: Either the 'git rev-parse HEAD' output or None
+    """
+    try:
+        git_hash = subprocess.check_output(['git', 'rev-parse', 'HEAD']).strip().decode('utf-8')
+    except Exception as e:
+        logging.info("Unable to infer git hash, error: %s", str(e))
+        git_hash = None
+    return git_hash
+
+
+def infer_triage_version():
+    return __version__
+
+
+def infer_installed_libraries():
+    """Attempt to infer the installed libraries by running pip freeze and formatting as a list
+
+    Returns: Either a list, or None
+    """
+    try:
```

Comment:
I believe this can be in-process:

```
if pip_freeze is not None:
    installed_libs = pip_freeze.freeze()
else:
    ...
```

Where we've imported that as:

```
try:
    try:
        from pip._internal.operations import freeze as pip_freeze
    except ImportError:  # pip < 10.0
        from pip.operations import freeze as pip_freeze
except ImportError:
    pip_freeze = None
```